client/landing/stepper/declarative-flow/internals/components/step-route/hooks/use-step-route-tracking/index.tsx

```diff
@@ -21,34 +20,33 @@ import {
 import { useSelector } from 'calypso/state';
 import { isRequestingSite } from 'calypso/state/sites/selectors';
 
-// Ensure that the selected site is fetched, if available. This is used for event tracking purposes.
-// See https://github.com/Automattic/wp-calypso/pull/82981.
-const useIsRequestingSelectedSite = ( siteSlugOrId: string | number, site: SiteDetails | null ) => {
-	return useSelector( ( state ) => site && isRequestingSite( state, siteSlugOrId ) );
+/**
+ * We wait for the site to be fetched before tracking the step route when a site ID/slug are defined in the params.
+ * This is to ensure that the site data is available for event tracking purposes.
+ * See `superProps`, `site_plan_id`, and https://github.com/Automattic/wp-calypso/pull/82981.
+ */
+const useHasRequestedSelectedSite = () => {
```

Comment:
This function works well, I just had a bit of a hard time reading the return of it.

What do you think if we make it simpler?

Example 

```js
const useHasRequestedSelectedSite = () => {
	const { site, siteSlugOrId } = useSiteData();

	if (!siteSlugOrId) {
		return true;
	}

	const isRequesting = useSelector((state) => isRequestingSite(state, siteSlugOrId));

	return !!site && !isRequesting;
};
```

I haven't test, but I believe it is doing the same and IMO more readable. This is totally optional.

Reply:
Thanks. It felt same way to me too. We can't do an early return though before `useSelector` call (can't call hooks conditionally). 

Let me know if anything comes to mind. I think before it was a bit worse with being scattered around (but we could bring that version back too) 🤔

Reply:
True it is inside a hook. 
I've asked "help" to our good friend gpt.
It suggested.

```js
const useHasRequestedSelectedSite = () => {
  const { site, siteSlugOrId } = useSiteData();

  // Always call the hook but adjust logic within the selector
  const isRequestingSelectedSite = useSelector((state) => {
    if (!site || !siteSlugOrId) {
      return false; // No request if site or siteSlugOrId is missing
    }
    return isRequestingSite(state, siteSlugOrId);
  });

  // Memoize the result to prevent unnecessary re-renders
  return useMemo(() => {
    if (!siteSlugOrId) {
      return true; // If no siteSlugOrId, assume requested
    }
    return !!site && !isRequestingSelectedSite;
  }, [site, siteSlugOrId, isRequestingSelectedSite]);
};

```
I've tested and works the same, plus I liked the useMemo here.
Again, this is totally optional.

Reply:
Interesting. Thanks for probing the AI for it! :D 

I feel memoizing anything here isn't all that effective. The value is true/false, so unless we are saving on processing (I think we are not), then it's probably unnecessary?

Reply:
Think it could be useful, preventing unnecessary re-renderings, needs better validation, though.

Reply:
I think memo here seems useless, especially as far as rerenders go.

The way I see it, memoizing has two main underlying purposes:

1. Avoid redoing expensive calculations
2. Ensure referential integrity on return values to avoid rerenders (this only concerns `Object` return values)

The return value here is a boolean `true`/`false`. There's no reference being recreated if this hook is called a thousand times and returns a thousand `true`. They are the same static value. Much like a string or a number or `null`/`undefined`.